#11 온보딩 페이지 구현

gyurim opened the pull request:
- 온보딩 3단계 페이지 구현
- NumberPicker로 선택한 값들(hours, minutes)은 추후 DB 생성 후 저장하도록 코드 변경 예정

gyurim committed:
#11 온보딩 페이지 구현

- 온보딩 3단계 페이지 구현
- NumberPicker로 선택한 값들(hours, minutes)은 추후 DB 생성 후 저장하도록 코드 변경 예정

JJJoonngg commented on NanDoChak/app/src/main/AndroidManifest.xml:
```diff
@@ -33,6 +33,9 @@
         </activity>
+        <activity android:name=".view.averagereadytime.AverageReadyTimeActivity"></activity>
+        <activity android:name=".view.earlyarrivedtime.EarlyArrivedTimeActivity"></activity>
+        <activity android:name=".view.plantime.PlanTimeActivity"></activity>
```
뒤에 불필요한 </activity> 태그는 빼도 될거같아요~!

JJJoonngg commented on NanDoChak/app/src/main/java/kr/co/west_gang/nan_do_chak/view/averagereadytime/AverageReadyTimeActivity.kt:
```diff
@@ -0,0 +1,71 @@
+        binding.lifecycleOwner = this
+        binding.viewModel = viewModel
+
+        logD(AppConfig.TAG_DEBUG, "AverageReadyTime Activity onCreate")
```
해당 코드는 나중에는 삭제해도 될거같아요~! 현재는 그냥 두는걸로 하죠

JJJoonngg commented on NanDoChak/app/src/main/java/kr/co/west_gang/nan_do_chak/view/averagereadytime/AverageReadyTimeActivity.kt:
```diff
@@ -0,0 +1,71 @@
+    private fun gotoEarlyArrivedTime(){
+        startActivity(
+            Intent(this, EarlyArrivedTimeActivity::class.java)
+        )
```
뒤에 finish 는 필요 없을까요?? activity 스택에 굳이 쌓을 필요가 없을거 같아서요!

gyurim replied:
넵! 우선은 뒤로가기 버튼 누르면 수정가능하게 구현했습니당

JJJoonngg commented on NanDoChak/app/src/main/java/kr/co/west_gang/nan_do_chak/view/earlyarrivedtime/EarlyArrivedTimeViewModel.kt:
```diff
@@ -0,0 +1,14 @@
+    val buttonClickEvent : LiveData<Unit> = _buttonClickEvent
+
+    fun onButtonClick(){
+        _buttonClickEvent.value = Unit
```
button 클릭시에 value 를 여기서 바로 가져가는건 어떨까요 따로 get 함수를 만들지 않아도 될거같아서 
따로 전달하는 부분이 없다면 여기서 지정해도 될거같네요

gyurim replied:
오 좋아요! 해당 코드로 변경하겠습니다

JJJoonngg commented on NanDoChak/app/src/main/java/kr/co/west_gang/nan_do_chak/view/plantime/PlanTimeActivity.kt:
```diff
@@ -0,0 +1,79 @@
+        planTimeMinutes = binding.planTimeMinutesPicker.value
+        planTime = valueOfPlanTimePicker[binding.planTimePicker.value]
+
+        logD(AppConfig.TAG_DEBUG, "hours: $planTimeHours / minutes: $planTimeMinutes / time: $planTime")
```
https://stackoverflow.com/questions/48368009/how-to-pass-numberpicker-value-onvaluechange-through-data-binding

위에도 언급한 부분인데 위의 방식대로 작업하는건 어떨까요!?!?

JJJoonngg commented on NanDoChak/app/src/main/res/layout/activity_average_ready_time.xml:
```diff
@@ -0,0 +1,99 @@
+        tools:context=".view.averagereadytime.AverageReadyTimeActivity">
+
+        <TextView
+            android:id="@+id/textView"
```
명칭이 모호해요 textView 가 어디서 어떤 textView 인지 좀 더 명확한 네이밍이 필요할 것 같습니다

JJJoonngg commented on NanDoChak/app/src/main/res/layout/activity_average_ready_time.xml:
```diff
@@ -0,0 +1,99 @@
+            android:layout_marginStart="16dp"
+            android:layout_marginTop="50dp"
+            android:letterSpacing="-0.01"
+            android:lineSpacingExtra="8sp"
```
해당부분들은 style 로 만들어서 진행하는 건 어떨까요!?!?

JJJoonngg commented on NanDoChak/app/src/main/res/layout/activity_average_ready_time.xml:
```diff
@@ -0,0 +1,99 @@
+            android:textColor="@color/on_boarding_text_color_black"/>
+
+        <ImageView
+            android:id="@+id/imageView"
```
네이밍 부탁드려요~!

JJJoonngg commented on NanDoChak/app/src/main/res/layout/activity_early_arrived_time.xml:
```diff
@@ -0,0 +1,100 @@
+        tools:context=".view.earlyarrivedtime.EarlyArrivedTimeActivity">
+
+        <TextView
+            android:id="@+id/textView"
```
네이밍 !

JJJoonngg commented on NanDoChak/app/src/main/res/layout/activity_early_arrived_time.xml:
```diff
@@ -0,0 +1,100 @@
+            android:textColor="@color/on_boarding_text_color_black"/>
+
+        <ImageView
+            android:id="@+id/imageView"
```
네이밍!

JJJoonngg commented on NanDoChak/app/src/main/res/layout/activity_early_arrived_time.xml:
```diff
@@ -0,0 +1,100 @@
+            android:lineSpacingExtra="8sp"
+            android:text="@string/selectEarlyArrivedTime"
+            android:textSize="22dp"
+            android:textStyle="bold"
```
스타일!

JJJoonngg commented on NanDoChak/app/src/main/res/layout/activity_main.xml:
```diff
@@ -16,6 +16,7 @@
         tools:context=".view.main.MainActivity">
 
         <TextView
+            android:id="@+id/textView"
```
네이밍!

JJJoonngg commented on NanDoChak/app/src/main/res/layout/activity_plan_time.xml:
```diff
@@ -0,0 +1,110 @@
+        tools:context=".view.plantime.PlanTimeActivity">
+
+        <TextView
+            android:id="@+id/textView"
```
네이밍!

JJJoonngg commented on NanDoChak/app/src/main/res/layout/activity_plan_time.xml:
```diff
@@ -0,0 +1,110 @@
+            android:textColor="@color/on_boarding_text_color_black"/>
+
+        <ImageView
+            android:id="@+id/imageView"
```
네이밍!

JJJoonngg commented on NanDoChak/app/src/main/res/values/strings.xml:
```diff
@@ -1,3 +1,15 @@
+    <string name="selectAverageReadyTime">친구야 안녕!\n평균 준비 시간을 선택해줘요</string>
+
+    <!--early arrived time-->
+    <string name="selectEarlyArrivedTime">친구 이름이 $유저네임$(이)구나!\n몇 분 일찍 도착할 계획일까?</string>
```
해당부분은 분리해서 유저 네임으로 바꿀 수 있게끔 하면 좋을거 같아요

gyurim committed:
#11 리뷰 반영

- data binding 방법으로 value 저장
- style 사용해 중복 xml 코드 삭제
- 변수명 수정
- 현재 activity 3개로 되어있는 코드를 추후 singleActivity&fragment 사용해 적용해보겠습니다.

gyurim committed:
Merge branch 'develop' into feature/#11

JJJoonngg commented on NanDoChak/app/src/main/java/kr/co/west_gang/nan_do_chak/view/plantime/PlanTimeViewModel.kt:
```diff
@@ -8,7 +8,18 @@ class PlanTimeViewModel : ViewModel() {
 
+    var planTimeHours = 0
+    var planTimeMinutes = 0
+    var planTimeMeridiem = ""
```
live data 이용하는걸로 수정해주세요

gyurim committed:
#11 리뷰 반영

- live data 사용

gyurim committed:
Merge branch 'feature/#11' of https://github.com/DDD-5/westgang-nandochak-android into feature/#11
